pyccel/codegen/printing/ccode.py

```diff
@@ -586,6 +602,68 @@ def function_signature(self, expr):
+        else:
+            base = expr.base
+        base = self._print(expr.base.label)
+        inds = list(expr.indices)
+        inds = inds[::-1]
+        base_shape = Shape(expr.base)
+        allow_negative_indexes = (isinstance(expr.base, IndexedVariable) and \
+                expr.base.internal_variable.allows_negative_indexes)
+        for i, ind in enumerate(inds):
+            if isinstance(ind, PyccelUnarySub) and isinstance(ind.args[0], Integer):
+                inds[i] = PyccelMinus(base_shape[i], ind.args[0])
+            else:
+                if isinstance(ind, Slice):
+                    #setting the slice start and end to their correct value if none is provided
+                    start = ind.start
+                    end = ind.end
+                    if ind.start is None:
+                        start = 0
+                    if ind.end is None:
+                        end = expr.base.shape[i]
+                    inds[i] = Slice.__new__(Slice, start, end)
+                #indices of indexedElement of len==1 shouldn't be a Tuple
+                if isinstance(ind, Tuple) and len(ind) == 1:
+                    inds[i].args = ind[0]
+                if allow_negative_indexes and not isinstance(ind, Integer):
+                    inds[i] = PyccelMod(ind, base_shape[i])
+        inds = [self._print(i) for i in inds]
+        #set dtype to the C struct types
+        dtype = self.find_in_dtype_registry(format(expr.dtype), expr.precision, array=True)
+        if expr.rank > 0:
+            return "array_slicing(%s, %s)" % (base, ", ".join(inds))
+        return "%s.nd_%s[get_index(%s, %s)]" % (base, dtype, base, ", ".join(inds))
+
+    def _print_Allocate(self, expr):
+        free_code = ''
+        #free the array if its already allocated
+        if  (expr.status != 'unallocated'):
+            free_code = 'if (%s.raw_data != NULL)\n' % self._print(expr.variable.name)
+            free_code += '{\nfree_array(%s);\n}\n' % self._print(expr.variable.name)
+        self._additional_imports.add('ndarrays')
+        shape = ", ".join(str(a) for a in expr.shape)
+        dtype = self.find_in_dtype_registry(format(expr.variable.dtype), expr.variable.precision, array=True)
+        shape_name = "shape_dummy_%s" % format(expr.__hash__()%991)
+        init_shape = "int %s[] = {%s};" % (shape_name, shape)
+        alloc_code = "{} = array_create({}, {}, nd_{});".format(expr.variable, len(expr.shape), shape_name, dtype)
+        return '{}{}\n{}'.format(free_code, init_shape, alloc_code)
+
+    def _print_Slice(self, expr):
+        # print(expr.args)
+        if expr.start is None or  isinstance(expr.start, Nil):
+            start = '-1'
+        else:
+            start = self._print(expr.start)
+        if (expr.end is None) or isinstance(expr.end, Nil):
+            end = '-1'
+        else:
+            end = self._print(expr.end)
+        return 'new_slice({}, {}, {})'.format(start, end, 1)
```

Comment:
Why do you not collect the slice step?

Reply:
its still not implemented in the slice class i will try to see if i can manage to add it

Reply:
Ah sorry I didn't realise it wasn't implemented. In that case I would leave it as it is. We can add steps in a different pull request as we will need to add them in fortran too when we do add them